server/src/main/java/com/cloud/template/TemplateManagerImpl.java

```diff
@@ -1215,7 +1215,7 @@ public TemplateInfo prepareIso(long isoId, long dcId, Long hostId, Long poolId)
             tmplt = _tmplFactory.getReadyBypassedTemplateOnPrimaryStore(isoId, poolId, hostId);
             bypassed = true;
         } else {
-            tmplt = _tmplFactory.getTemplate(isoId, DataStoreRole.Image, dcId);
+            tmplt = _tmplFactory.getReadyTemplateOnImageStore(isoId, dcId);
```

Comment:
looking at the implementations both calls are functionaly similar and should be unified. `getReadyTemplateOnImageStore` has `DataStoreRole.Image` hard-coded, while `getTemplate` gets it as a parameter. I am not sure if this will help anything.

Reply:
@DaanHoogland 
yes, they both look for Templates /isos on image stores.
The difference is, `getTemplate` do not have the check on State

Reply:
`findByTemplateZoneReady` vs `findByTemplateZone`, i see. makes sense. I still think this should be implemented more prozaic but sensible improvement :+1:

Reply:
let's merge, improvements aside this should help indeed.

Reply:
good, let me merge it
thanks for review @DaanHoogland